lib/ansible/plugins/inventory/aws_ssm.py

```diff
@@ -0,0 +1,213 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: aws_ssm
+    plugin_type: inventory
+    author:
+      - Andrew Hibbert <andrew_hibbert@hotmail.com>
+    requirements:
+    - boto3
+    - botocore
+    short_description: AWS SSM inventory source
+    description:
+      - Fetch parameters via path
+      - Add groups, hosts and variables based on path
+    options:
+      plugin:
+        description: token that ensures this is a source file for the 'aws_ssm' plugin.
+        required: True
+        choices: ['aws_ssm']
+      path:
+        description: Path to search for in ssm
+        required: True
+      region:
+        description: Region to retrieve SSM parameter from
+        required: True
+      decrypt:
+        description: Whether to decrypt secrets or not
+        required: False
+        default: True
+'''
```

Comment:
the credentials/cache/constructed options are undocumented.

```suggestion
    extends_documentation_fragment:
        - inventory_cache
        - constructed
        - aws_credentials
'''
```

Reply:
Made this change

Reply:
I have made this change